src/Checkbox.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import Icon from './Icon'
+import Flex from './Flex'
+import Box from './Box'
+import theme from './theme'
+
+class Checkbox extends React.Component {
+  constructor({ value, isChecked }) {
+    super()
+    this.state = {
+      value,
+      isChecked
+    }
+  }
+
+  handleClick() {
+    const { isDisabled, name, value, checkAction } = this.props
+    const { isChecked } = this.state
+    if (isDisabled) {
+      return
+    }
+
+    checkAction(name, value, !isChecked)
+    this.setState({ isChecked: !isChecked })
+  }
+
+  render() {
+    const { name, id, isDisabled, value, size, fontSize } = this.props
+
+    const currentTheme = this.props.theme || theme
+
+    const CurrentComponent = this.state.isChecked
+      ? CheckedCheckbox
+      : StyledCheckbox
+    return (
+      <CurrentComponent
+        isChecked={this.state.isChecked}
+        isDisabled={isDisabled}
+        size={size}
+        theme={currentTheme}
+        data-test="design-system-checkbox"
+        onClick={() => this.handleClick()}
+      >
+        {this.state.isChecked ? (
+          <Icon
+            name="check"
```

Comment:
I *think* @odmln intended this component and the Radio component to use two different icons for the checked and unchecked states – `box` and `boxChecked` I believe... https://pricelinelabs.github.io/design-system/?selectedKind=Icon&selectedStory=Icons&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel